reflred/backgroundfield.py

```diff
@@ -0,0 +1,256 @@
+    s4 = list()
+    v = list()
+    dv = list()
+    Qz_target = list()
+
+    # Check for slit distances, otherwise use values from data file
+    LS3 = back[0].slit3.distance if not np.isinf(back[0].slit3.distance) else LS3
+    LS4 = back[0].slit4.distance if not np.isinf(back[0].slit4.distance) else LS4
+    L4D = LSD - LS4
+
+    # Extract relevant quantities from inputs
+    for b in back:
+
+        ai.append(b.sample.angle_x)
+        Qz_target.append(b.Qz_target if not np.all(np.isnan(b.Qz_target)) else 4*np.pi/b.detector.wavelength*np.sin(np.radians(b.sample.angle_x)))
+        af.append(b.detector.angle_x - b.sample.angle_x)
+        s3.append(b.slit3.x)
+        s4.append(b.slit4.x if not np.all(np.isinf(b.slit4.x)) else 25. * np.ones(b.v.shape))
+        v.append(b.v)
+        dv.append(b.dv)
+
+    # Concatenate and filter results based on the Qcutoff parameter
+    Qz_target = np.concatenate(Qz_target)
+    ai = np.radians(np.concatenate(ai))
+    af = np.radians(np.concatenate(af))
+    crit = (Qz_target > Qcutoff) & (af > 0.)
+    ai = ai[crit]
+    af = af[crit]
+    s3 = np.concatenate(s3)[crit]
+    s4 = np.concatenate(s4)[crit]
+    v = np.concatenate(v)[crit]
+    dv = np.concatenate(dv)[crit]
+
+    # Calculate the detector footprints and use maximum value relative to sample size maxF
+    Fd = detector_footprint(af,
+                            s3,
+                            s4,
+                            LS3,
+                            LS4 - LS3)
+
+    Fd = np.min((Fd, maxF*np.ones(Fd.shape)), axis=0)
+
+    # Calculate the detector solid angle
+    SA = fractional_solid_angle(s4, LS4, L4D, HD)
+
+    # Define fit function and Jacobian
+    def minfunc(pars):
+
+        scale, epsD = pars
+        return (background_reservoir(ai,
+                                    epsD,
+                                    af,
+                                    epssi,
+                                    Fd,
+                                    scale)[0] * SA - v)/dv
+
+    def jacfunc(pars):
+
+        scale, epsD = pars
+        return (background_reservoir_jac(ai, epsD, af, epssi, Fd, scale)*np.tile(SA/dv, (len(pars), 1))).T
+
+    # Condition the fit if the scale factor is not included in the fit, then perform the fit
+    if not fit_scale:
+        pout, pcov = so.leastsq(lambda epsD: minfunc([scale_value, epsD]), epsD0, Dfun=lambda epsD: jacfunc([1.0, epsD])[:,1], full_output=True, ftol=1e-15,
+                                xtol=1e-15)[:2]
+        lenpout = len(pout)
+        pout = np.array([scale_value, pout])
+        pcov = np.array([[0.0, 0.0],[0.0, pcov[0]]])
+    else:
+        pout, pcov = so.leastsq(minfunc, np.array([scale_value, epsD0]), Dfun=jacfunc, full_output=True, ftol=1e-15, xtol=1e-15)[:2]
+        lenpout = len(pout)
+
+    # Calculate fitting error (not used) and chi-squared (only reported)
+    perr, chisq = np.sqrt(np.diag(pcov)), np.sqrt(np.sum(minfunc(pout)**2) / (len(v) - lenpout))
```

Comment:
Why the sqrt() in the calculation of chisq?

Reply:
Great catch, thanks. Sqrt removed.